crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,180 @@
+use crate::{
+    types::{block_identifier::BlockIdentifier, receipt::RpcLog},
+    RpcErr, RpcHandler,
+};
+use ethereum_rust_core::{H160, H256};
+use ethereum_rust_storage::Store;
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashSet;
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+pub enum AddressFilter {
+    Single(H160),
+    Many(Vec<H160>),
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+pub enum TopicFilter {
+    Topic(Option<H256>),
+    Topics(Vec<Option<H256>>),
+}
+
+#[derive(Debug)]
+pub struct LogsRequest {
+    /// The oldest block from which to start
+    /// retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub from_block: BlockIdentifier,
+    /// Up to which block to stop retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub to_block: BlockIdentifier,
+    /// The addresses from where the logs origin from.
+    pub address_filters: Option<AddressFilter>,
+    /// Which topics to filter.
+    pub topics: Vec<TopicFilter>,
+}
+impl RpcHandler for LogsRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<LogsRequest, RpcErr> {
+        match params.as_deref() {
+            Some([param]) => {
+                let param = param.as_object().ok_or(RpcErr::BadParams)?;
+                let from_block = {
+                    if let Some(param) = param.get("fromBlock") {
+                        BlockIdentifier::parse(param.clone(), 0)?
+                    } else {
+                        BlockIdentifier::latest()
```

Comment:
is this the actual default as per the spec?

Reply:
<img width="1211" alt="image" src="https://github.com/user-attachments/assets/b8f8f8e2-a6b2-4481-ba5e-ba998c52268e">


It it's not, even more so, the spec seems to only expect a hex number, should I restrict the parameters to that?
I'll start removing the default value and return an error in its place.

Reply:
Also, there doesn't seem to be a default value [here](https://github.com/ethereum/execution-apis/blob/4aca1d7a3e5aab24c8f6437131289ad386944eaa/src/schemas/filter.yaml#L12)

Reply:
I've removed the default 'Block Latest' value and instead I'm returning a 'MissingParam' error.